demos/CMakeLists.txt

```diff
@@ -176,8 +176,8 @@ macro(ie_add_sample)
         target_include_directories(${IE_SAMPLE_NAME} PRIVATE ${IE_SAMPLE_INCLUDE_DIRECTORIES})
     endif()
 
-    target_link_libraries(${IE_SAMPLE_NAME} PRIVATE ${OpenCV_LIBRARIES} ${InferenceEngine_LIBRARIES}
-                                                    ${IE_SAMPLE_DEPENDENCIES} common gflags)
+    target_link_libraries(${IE_SAMPLE_NAME} PRIVATE ${TBB_IMPORTED_TARGETS} ${OpenCV_LIBRARIES} ${InferenceEngine_LIBRARIES}
+                                                    ngraph::ngraph ${IE_SAMPLE_DEPENDENCIES} common gflags)
```

Comment:
Why do we need `nGraph` dependency for sample?
Do samples work with nGraph function and it causes this dependency?

Reply:
@ilyachur The simple answer is, if you don't link ngraph sample will not compile (link). And answer to your question here : "I don't understand at the current moment why do you need TBB and nGraph. Demos don't touch it, Inference Engine has a private dependency on these components. In general if you use the right environment all should work.
Am I wrong?" Respectfully @ilyachur , yes you are wrong. I can prove it to you if you set up some time with me. Or you can try it yourself. Try to build demo against **_a released version of OpenVINO_** (for instance 2020.4) and do not run setupvars.sh. Try building the demos now **_without running setupvars.sh_** .  Try again in the oecore (cross-compiled) environment. Does it work ? You will get errors with ngraph and/or TBB.

Reply:
@shubha-ramani who said that we must support scenario when setupvars.sh is not used?

Reply:
As I said, if you use the right environment all should work. But if you don't use **setupvars.sh** you can have incorrect path. And this causes your issues, because nGraph or TBB cannot be found in the environment path.